modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/fragment/renderer/SpacesSectionFragmentRenderer.java

```diff
@@ -144,6 +145,57 @@ public void render(
 		}
 	}
 
+	private Collection<AssetLibrary> _getItems(
+		Page<AssetLibrary> assetLibrariesPinnedByMePage,
+		Page<AssetLibrary> assetLibrariesPage) {
+
+		List<AssetLibrary> assetLibrariesPinnedByMePageItems = new ArrayList<>(
+			assetLibrariesPinnedByMePage.getItems());
+
+		if (assetLibrariesPinnedByMePageItems.size() == 5) {
+			return assetLibrariesPinnedByMePageItems;
+		}
+
+		List<AssetLibrary> assetLibrariesPageItems = new ArrayList<>(
+			assetLibrariesPage.getItems());
+
+		if (assetLibrariesPinnedByMePageItems.isEmpty()) {
+			return assetLibrariesPageItems.subList(0, 5);
+		}
+
+		List<AssetLibrary> items = assetLibrariesPinnedByMePageItems;
+
+		for (AssetLibrary assetLibraryItem : assetLibrariesPageItems) {
+			items.add(assetLibraryItem);
+
+			if (items.size() == 5) {
+				return items;
+			}
+		}
+
+		return assetLibrariesPageItems;
+	}
+
+	private Page<AssetLibrary> _getPage(
+			AssetLibraryResource assetLibraryResource)
+		throws Exception {
+
+		Page<AssetLibrary> assetLibrariesPinnedByMePage =
+			assetLibraryResource.getAssetLibrariesPinnedByMePage();
+
+		Page<AssetLibrary> assetLibrariesPage =
+			assetLibraryResource.getAssetLibrariesPage(
+				null, null, null,
+				Pagination.of(
+					1, (int)(5 + assetLibrariesPinnedByMePage.getPageSize())),
+				null);
```

Comment:
You should only invoke `getAssetLibrariesPage` iff `assetLibrariesPinnedByMePage.size() < 5`. Also,  if `assetLibrariesPinnedByMePage.size() > 5` return those 5 first elements and avoid the extra call as well.

When you need to do the extra call, then ask for the minimum number of elements that you need (5), and not more.

Reply:
After a private conversation, the extra call is required for the (global) count of spaces.